src/changelog/comment_changes.rs

```diff
@@ -305,6 +334,12 @@ impl Logic {
                                     if stop_at == oid {
                                         break;
                                     }
+                                } else if let Some(stop_at) =
+                                    self.stop_at_tag_oid
+                                {
+                                    if stop_at == oid {
+                                        break;
+                                    }
```

Comment:
The new stop condition might be reached earlier than the explicit commit number and vice versa.  Whichever condition is met first will cause Aeruginous to generate the intended fragment from the already harvested changes.

Reply:
Have you tried the new `is_some_and()` function in rust? It could make the code a little bit more readable like:
```rust
} else if self.stop_at_tag_oid.is_some_and(|stop_at| stop_at == oid) {
   break;
}
```

Reply:
Thank you for this awesome suggestion, @AmmarAbouZor!  I just applied it around the respective region.  Please feel free to submit further suggestions and comments.